types/multicoin-address-validator/index.d.ts

```diff
@@ -0,0 +1,25 @@
+// Type definitions for multicoin-address-validator 0.5
+// Project: https://github.com/christsim/multicoin-address-validator
+// Definitions by: Kyle McLean <https://github.com/kylemclean>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace WAValidator;
+
+export function validate(address: string, currencyNameOrSymbol?: string, opts?: ValidateOpts | string): boolean;
+export function getCurrencies(): Currency[];
+export function findCurrency(symbol: string): Currency | null;
+
+export interface Validator {
+    isValidAddress: (address: string, currency: Currency, opts: ValidateOpts) => boolean;
+}
+
+export interface Currency {
+    name: string;
+    symbol: string;
+    validator: Validator;
+    [key: string]: any;
```

Comment:
```ts
    [key: string]: any;
```
this could be obviously more specific, as there are closed list of props used in currencies and limited list of possible props values. Those would be optional though.

Reply:
My reasoning behind not including the additional properties was that some of them (such as `regexp`, `regex`, `hashFunction`, and `iAddressTypes`) are used in only a couple of currencies, and there are some inconsistencies in the types. For example, I noticed that the `addressTypes` property of Tron contains `number[]` whereas the other currencies with this property have `string[]`.

I think that most of the properties are only meant to be used in the currency's private implementations of `isValidAddress`, though perhaps declaring their types would still be useful.